### PR TITLE
Add teleporter logs

### PR DESCRIPTION
**Pull request by Chief-Engineer**
## About the PR 
- Add Teleported and EntityDelete log types
- Log portal opened
- Log portal closed
- Log teleportation

**Media**

- [X] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

### CODE REVIEW

File: Content.Server/Teleportation/HandTeleporterSystem.cs [hunk 1]
```diff
@@ -98,18 +101,29 @@ private void HandlePortalUpdating(EntityUid uid, HandTeleporterComponent compone
             var timeout = EnsureComp<PortalTimeoutComponent>(user);
             timeout.EnteredPortal = null;
             component.FirstPortal = Spawn(component.FirstPortalPrototype, Transform(user).Coordinates);
+            _adminLogger.Add(LogType.EntitySpawn, LogImpact.Low, $"{ToPrettyString(user):player} opened {ToPrettyString(component.FirstPortal.Value)} at {Transform(component.FirstPortal.Value).Coordinates} using {ToPrettyString(component.Owner)}");
```
Comment (metalgearsloth) [inline Content.Server/Teleportation/HandTeleporterSystem.cs:104]:
```suggestion
            _adminLogger.Add(LogType.EntitySpawn, LogImpact.Low, $"{ToPrettyString(user):player} opened {ToPrettyString(component.FirstPortal.Value)} at {Transform(component.FirstPortal.Value).Coordinates} using {ToPrettyString(uid)}");
```

File: Content.Server/Teleportation/HandTeleporterSystem.cs [hunk 2]
```diff
@@ -98,18 +101,29 @@ private void HandlePortalUpdating(EntityUid uid, HandTeleporterComponent compone
...
             _audio.PlayPvs(component.NewPortalSound, uid);
         }
         else if (component.SecondPortal == null)
         {
             var timeout = EnsureComp<PortalTimeoutComponent>(user);
             timeout.EnteredPortal = null;
             component.SecondPortal = Spawn(component.SecondPortalPrototype, Transform(user).Coordinates);
+            _adminLogger.Add(LogType.EntitySpawn, LogImpact.Low, $"{ToPrettyString(user):player} opened {ToPrettyString(component.SecondPortal.Value)} at {Transform(component.SecondPortal.Value).Coordinates} linked to {ToPrettyString(component.FirstPortal!.Value)} using {ToPrettyString(component.Owner)}");
```
Comment (metalgearsloth) [inline Content.Server/Teleportation/HandTeleporterSystem.cs:112]:
```suggestion
            _adminLogger.Add(LogType.EntitySpawn, LogImpact.Low, $"{ToPrettyString(user):player} opened {ToPrettyString(component.SecondPortal.Value)} at {Transform(component.SecondPortal.Value).Coordinates} linked to {ToPrettyString(component.FirstPortal!.Value)} using {ToPrettyString(uid)}");
```

File: Content.Server/Teleportation/HandTeleporterSystem.cs [hunk 3]
```diff
@@ -98,18 +101,29 @@ private void HandlePortalUpdating(EntityUid uid, HandTeleporterComponent compone
...
         else
         {
+            // Logging
+            var portalStrings = "";
+            portalStrings += ToPrettyString(component.FirstPortal!.Value);
+            if (portalStrings != "")
+                portalStrings += " and ";
+            portalStrings += ToPrettyString(component.SecondPortal!.Value);
+            if (portalStrings != "")
+                _adminLogger.Add(LogType.EntityDelete, LogImpact.Low, $"{ToPrettyString(user):player} closed {portalStrings} with {ToPrettyString(component.Owner)}");
```
Comment (metalgearsloth) [inline Content.Server/Teleportation/HandTeleporterSystem.cs:125]:
```suggestion
                _adminLogger.Add(LogType.EntityDelete, LogImpact.Low, $"{ToPrettyString(user):player} closed {portalStrings} with {ToPrettyString(uid)}");
```

File: Content.Server/Teleportation/PortalSystem.cs [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+
+public sealed class PortalSystem : SharedPortalSystem
+{
+    [Dependency] private readonly ISharedAdminLogManager _adminLogger = default!;
+
+    protected override void LogTeleport(EntityUid portal, EntityUid subject, EntityCoordinates source,
+        EntityCoordinates target)
+    {
+        if (HasComp<MindComponent>(subject))
+            _adminLogger.Add(LogType.Teleport, LogImpact.Low, $"{ToPrettyString(subject):player} teleported via {ToPrettyString(portal)} from {source} to {target}");
```
Comment (metalgearsloth) [inline Content.Server/Teleportation/PortalSystem.cs:17]: Isn't there a shared interface for this

File: Content.Shared.Database/LogType.cs [hunk 5]
```diff
@@ -81,4 +81,6 @@ public enum LogType
     Stamina = 76,
     EntitySpawn = 77,
     Container = 78,
+    Teleport = 79,
+    EntityDelete = 80,
```
Comment (metalgearsloth) [inline Content.Shared.Database/LogType.cs:85]: EntityDelete is too generic, I'd just make it Portal.

**Review by metalgearsloth — CHANGES_REQUESTED** [review]

File: Content.Shared.Database/LogType.cs (hunk 5 above)
Comment (mirrorcult) [reply Content.Shared.Database/LogType.cs]: entitydelete is fine i think but its weird if literally nothing else uses it

File: Content.Server/Teleportation/PortalSystem.cs (hunk 4 above)
Comment (mirrorcult) [reply Content.Server/Teleportation/PortalSystem.cs:17]: put a TODO on this that it should get moved to shared at some point

**Review by mirrorcult — CHANGES_REQUESTED** [review]

**Review by DrSmugleaf — APPROVED** [review]

**Pull request merged**